conans/client/tools/files.py

```diff
@@ -374,3 +375,48 @@ def remove_files_by_mask(directory, pattern):
                 os.unlink(fullname)
                 removed_names.append(os.path.relpath(fullname, directory))
     return removed_names
+
+
+def fix_symlinks(conanfile, raise_if_error=False):
+    """ Fix the symlinks in the conanfile.package_folder: make symlinks relative and remove
+        those links to files outside the package (it will print an error, or raise
+        if 'raise_if_error' evaluates to true).
+    """
+    offending_files = []
+
+    def work_on_element(dirpath, element, token):
+        fullpath = os.path.join(dirpath, element)
+        if os.path.islink(fullpath):
+            link_target = os.readlink(fullpath)
+            if link_target in ['/dev/null', ]:
+                return
+
+            link_abs_target = os.path.join(dirpath, link_target)
+            link_rel_target = os.path.relpath(link_abs_target, conanfile.package_folder)
+            if link_rel_target.startswith('..') or os.path.isabs(link_rel_target):
+                offending_file = os.path.relpath(fullpath, conanfile.package_folder)
+                offending_files.append(offending_file)
+                conanfile.output.error("{token} '{item}' links to a {token} outside the package, "
+                                       "it's been removed.".format(item=offending_file, token=token))
+                os.unlink(fullpath)
```

Comment:
this is nice that we output some error message(s) here, but we never return status code to the caller. so, for instance, conan will unable to return appropriate exit code to the user.

Reply:
There is a `raise_if_error` argument to the `tool`. It will raise for any of these errors. Returning from tools this kind of information could be something to add to here https://github.com/conan-io/conan/issues/7105 and reach a consensus.